src/components/views/voip/PipView.tsx

```diff
@@ -328,9 +335,37 @@ class PipView extends React.Component<IProps, IState> {
         this.setState({ showWidgetInPip, persistentWidgetId, persistentRoomId });
     }
 
+    private createVoiceBroadcastPreRecordingPipContent(
+        voiceBroadcastPreRecording: VoiceBroadcastPreRecording,
+    ): CreatePipContent {
+        return ({ onStartMoving }) => <div onMouseDown={onStartMoving}>
+            <VoiceBroadcastPreRecordingPip
+                voiceBroadcastPreRecording={voiceBroadcastPreRecording}
+            />
+        </div>;
+    }
+
+    private createVoiceBroadcastRecordingPipContent(
+        voiceBroadcastRecording: VoiceBroadcastRecording,
+    ): CreatePipContent {
+        return ({ onStartMoving }) => <div onMouseDown={onStartMoving}>
+            <VoiceBroadcastRecordingPip
+                recording={voiceBroadcastRecording}
+            />
+        </div>;
+    }
+
     public render() {
         const pipMode = true;
-        let pipContent;
+        let pipContent: CreatePipContent | null = null;
+
+        if (this.props.voiceBroadcastPreRecording) {
```

Comment:
moving this up here changes the functionality to give precedence to `this.state.primaryCall` and `this.state.showWidgetInPip` over `this.props.voiceBroadcastPreRecording` and `this.props.voiceBroadcastRecording`. Is that deliberate?

Reply:
Yes. For all new pip views I've added tests to check this.